packages/hdwallet-native/src/bitcoin.ts

```diff
@@ -270,11 +270,17 @@ export function MixinNativeBTCWallet<TBase extends core.Constructor<NativeHDWall
           try {
             const inputData = this.buildInput(coin, input);
 
-            psbt.addInput({
+            let inputDataFinal:any = {
               hash: input.txid,
               index: input.vout,
               ...inputData,
-            });
+            }
+            if(coin.toLowerCase() === "bitcoincash"){
```

Comment:
I would add this logic into the `buildInput` function and type the return properly to accept this feild as well.